x/ibc/07-tendermint/consensus_state.go

```diff
@@ -1,16 +1,20 @@
 package tendermint
 
 import (
+	"time"
+
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
 	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
 	commitment "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment"
+	tmtypes "github.com/tendermint/tendermint/types"
 )
 
 // ConsensusState defines a Tendermint consensus state
 type ConsensusState struct {
-	Root             commitment.RootI `json:"root" yaml:"root"`
-	ValidatorSetHash []byte           `json:"validator_set_hash" yaml:"validator_set_hash"`
+	Timestamp    time.Time             `json:"timestamp" yaml:"timestamp"`
+	Root         commitment.RootI      `json:"root" yaml:"root"`
+	ValidatorSet *tmtypes.ValidatorSet `json:"validator_set" yaml:"validator_set"`
```

Comment:
Why from hash to the full valset? Did we already discuss this and I forgot?

Reply:
See the ICS PR linked on the description